### DISCUSSION TITLE
LDPC 5g

### DISCUSSION THREAD

**Discussion by MarHajj** [NVlabs/sionna#50] [Q&A] [answered]
Hi,
I am working on a goal to simulate BER function of Eb/N0 using external data channel (measured and/or simulated).
I am using 1 transmitter and 1 receiver with one antenna for transmitting and receiving (SISO).

My OFDM ressource grid is :

ressourceGrid(num_ofdm_symbols=14,
                  fft_size=1024,
                  subcarrier_spacing=120e3,
                  num_tx=1,
                  num_streams_per_tx=1,
                  cyclic_prefix_length=72,
                  num_guard_carriers=[115,116],
                  dc_null=True,
                  pilot_pattern="kronecker",
                  pilot_ofdm_symbol_indices=[2,11])
with a code rate R = 0.36 (3GPP table 5.1.3.1-2 of the norme 38.214 (section 5.1.3)) and a 16 QAM (4 bits).

The code provide an error due to the number of information bits k is too large (attached screenshot)
Note that :
n = nbre of data symboles x nbre of qam bits
k = n x code rate

In this case can i segment my data into a packet of length less than 8448 or i have just to reduce my fft size ?

Thank you

Marwan EL HAJJ
![ldpc error](https://user-images.githubusercontent.com/112489064/187945972-d7c22848-4ae9-4e3e-bdea-c550ffe3dcac.png)

**Comment by jhoydis** [comment, marked as answer]
Hi Marwan,

Indeed, the code length in 5G is limited to 8448. Since Sionna does not provide transport block segmentation, you can do one of the following:

- Reduce the FFT size
- Reduce the number of OFDM symbols
- Concatenate multiple codewords
